src/mixins/object_geometry.mixin.ts

```diff
@@ -674,10 +675,7 @@ export class ObjectGeometry extends ObjectOrigin {
    * @return {TMat2D}
    */
   getViewportTransform(): TMat2D {
-    if (this.canvas && this.canvas.viewportTransform) {
-      return this.canvas.viewportTransform;
-    }
-    return iMatrix.concat() as TMat2D;
+    return this.canvas?.viewportTransform || (iMatrix.concat() as TMat2D);
```

Comment:
i m not sure we are transpiling those yet.
But the issue is that if undefined is not involved those becomes longer.
`if typeof canvas !== 'undefined'` rather than a simple `if this.canvas`

Reply:
please elaborate
I lack the knowledge

Reply:
What is the diff between this syntax and `obj?.[key]` ?

Reply:
none, but where we use obj?.[key] we were already looking for undefined ( ex: the overrideStyle object ). Here we want just to check if canvas is falsy in any way.

```
obj.first?.second

const temp = obj.first;
const nestedProp =
  temp === null || temp === undefined ? undefined : temp.second;
```

If we land on supported browsers that do not support the `?.` all those equality check get added in the codebase. I just don't like to think all that stuff gets added in.

Not a big issue anyway